distance-method/distance_method.py

```diff
@@ -823,22 +916,44 @@ def distance_method_main(query_proteome_fp,
         sys.stdout.write("\nRunning BLASTp ..\n")
     hits = {}
 
-    for ext in extensions:
-        for _file in glob("%s/*%s" % (target_proteomes_dir, ext)):
-            # launch BLASTp
-            alignments_fp = launch_blast(
-                query_proteome_fp=query_proteome_fp,
-                ref_fp=_file,
-                working_dir=working_dir,
-                e_value=e_value,
-                threads=threads,
-                debug=debug)
-
-            # generate a dictionary of orthologous genes
-            parse_blast(alignments_fp=alignments_fp,
-                        hits=hits,
-                        gene_map=gene_map,
+    # tabular alignments provided
+    if tabular_alignments_fp is not None:
+        # generate a dictionary of orthologous genes
+        parse_blast(alignments_fp=tabular_alignments_fp,
+                    hits=hits,
+                    gene_map=gene_map,
+                    debug=debug)
+    # tabular alignments to be created
+    else:
+        for ext in extensions:
+            for _file in glob("%s/*%s" % (target_proteomes_dir, ext)):
```

Comment:
Will it be easier to read the file system once and then filter the results by ext? This can be an issue is the user keeps adding files while this is running.

For example:
files = [f for ext in extensions for f in glob("%s/*%s" % (target_proteomes_dir, ext))]